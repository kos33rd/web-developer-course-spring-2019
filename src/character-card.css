.card {
    margin-top: 24px;
}

.media {
    height: 250px;
}
