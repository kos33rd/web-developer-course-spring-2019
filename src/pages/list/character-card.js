import React from 'react';
import Grid from '@material-ui/core/Grid';
import Card from '@material-ui/core/Card';
import CardActions from '@material-ui/core/CardActions';
import CardContent from '@material-ui/core/CardContent';
import CardMedia from '@material-ui/core/CardMedia';
import Typography from '@material-ui/core/Typography';
import Button from '@material-ui/core/Button';

import style from './character-card.css'

export const CharacterCard = ({ character, handleDetailsClick }) => (
    <Grid item sm={6} md={4} lg={3}>
        <Card className={style.card}>
            <CardMedia
                className={style.media}
                image={character.image}
                title="Avatar"
            />
            <CardContent>
                <Typography gutterBottom variant="h5" component="h2">
                    {character.name}
                </Typography>
                <Typography>
                    Origin: {character.origin.name}
                </Typography>
            </CardContent>
            <CardActions  className={style.bg}>
                <Button size="small" color="primary" onClick={handleDetailsClick}>
                    <span className={'class-name'}>Details</span>
                </Button>
            </CardActions>
        </Card>
    </Grid>
)
