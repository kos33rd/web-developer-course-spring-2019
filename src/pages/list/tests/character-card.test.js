import React from 'react'
import { CharacterCard } from '../character-card'

import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });


import { shallow, mount, render } from 'enzyme';

test('Grid renders as container', () => {

    const shallowComponent = shallow(<CharacterCard character={{origin: ''}} handleDetailsClick={() =>{}} />)
    const mountComponent = mount(<CharacterCard character={{origin: ''}} handleDetailsClick={() =>{}} />)
    expect(shallowComponent).toBeDefined()
    expect(shallowComponent.debug()).toMatchSnapshot()

    // Test lifecycle method calls and all possible component states
    // console.log(shallowComponent.debug())
    // console.log(mountComponent.debug())

})
