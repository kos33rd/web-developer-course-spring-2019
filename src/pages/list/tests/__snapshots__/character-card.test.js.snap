// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grid renders as container 1`] = `
"<Grid item={true} sm={6} md={4} lg={3} direction=\\"row\\">
  <WithStyles(Card) className=\\"card\\">
    <WithStyles(CardMedia) className=\\"media\\" image={[undefined]} title=\\"Avatar\\" />
    <WithStyles(CardContent)>
      <WithStyles(Typography) gutterBottom={true} variant=\\"h5\\" component=\\"h2\\" />
      <WithStyles(Typography)>
        Origin: 
      </WithStyles(Typography)>
    </WithStyles(CardContent)>
    <WithStyles(CardActions) className=\\"bg\\">
      <WithStyles(Button) size=\\"small\\" color=\\"primary\\" onClick={[Function: handleDetailsClick]}>
        <span className=\\"class-name\\">
          Details
        </span>
      </WithStyles(Button)>
    </WithStyles(CardActions)>
  </WithStyles(Card)>
</Grid>"
`;
